### ISSUE TITLE
PI0 LIBERO_BASE: PCC drop to 0.81 due to layerwise degradation across 18 Gemma decoder layers in paligemma language model forward

### ISSUE THREAD

**Issue by ashokkumarkannan1** [tenstorrent/tt-xla#3786]
### Describe the bug

The PI0 model (LIBERO_BASE variant) reports a whole-model PCC of 0.81 when comparing CPU vs TT device outputs. Through progressive slicing of the model, the root cause has been narrowed down to layerwise PCC degradation across the 18 Gemma decoder layers inside the paligemma language model forward pass.

### Slicing results:
- embed_prefix (SigLIP image embedding + language token embedding): passes -- all outputs match between CPU and TT
- Mask computation ops (make_att_2d_masks, cumsum, _prepare_attention_masks_4d): passes with PCC 1.0 (after bool→int64 fix)
- paligemma_with_expert.forward (prefix path, which internally calls paligemma.language_model.forward → GemmaModel.forward): PCC drops to ~0.98 on both hidden states and past key values
- Individual decoder layers (tested in sanity with 1, 2, 5, 10, 18 layers using saved inputs): each passes with PCC ≥ 0.99 for both hidden states and PKV.

This means each decoder layer individually meets the 0.99 PCC threshold, but when all 18 layers run sequentially inside the full GemmaModel.forward, the small per-layer errors compound. The paligemma forward sanity test reproduces the ~0.98 PCC drop, which then propagates through the 10 denoise steps in sample_actions, further degrading the final output to 0.81.

| Component | Sanity test | PCC | Status |
|---|---|---|---|
| `embed_prefix` (SigLIP image + language token embedding) | — | 1.0 | ✅ Pass |
| Mask ops (`make_att_2d_masks`, `cumsum`, `_prepare_attention_masks_4d`) | — | 1.0 | ✅ Pass (after bool→int64 fix) |
| `GemmaRMSNorm` alone (final norm after all decoder layers) | `test_gemma_norm.py` | **1.0** | ✅ Pass |
| `GemmaDecoderLayer` × 18 (saved inputs) | `test_pi_0_gemma_decoder_layers.py` | ≥ 0.99 | ✅ Pass |
| `paligemma_with_expert.forward` (full forward = 18 layers + norm) | `test_pi_0_paligemma_forward.py` | **~0.98** | ❌ Drop |
| Full model (prefix + 10× denoise) | — | **0.81** | ❌ Fail |

### Call chain where the drop originates:
```
PI0Policy.forward
  → PI0Pytorch.sample_actions
    → paligemma_with_expert.forward (prefix path)
      → paligemma.language_model.forward  (GemmaModel.forward)
        → 18x GemmaDecoderLayer  ← layerwise degradation here
    → 10x denoise_step  ← compounds the ~0.98 drop further to 0.81
```

### Steps to reproduce the issue

`git checkout akannan/pi_0_libero_drop`

- Run the paligemma forward sanity (reproduces ~0.98 PCC drop):
`pytest -svv tests/torch/models/pi_0/test_pi_0_paligemma_forward.py`

- Run the Gemma decoder layers sanity (shows individual layers pass):
`pytest -svv tests/torch/models/pi_0/test_pi_0_gemma_decoder_layers.py`

- Run the Gemma norm sanity (shows norm alone passes with PCC = 1.0):
`pytest -svv tests/torch/models/pi_0/test_gemma_norm.py`

-  Saved inputs are checked into the branch under `pi_0_libero_base_saved_inputs/` and logs are attached to the [branch](https://github.com/tenstorrent/tt-xla/tree/akannan/pi_0_libero_drop).

### Expected behavior

Paligemma forward sanity with Passing PCC>0.99

**Comment by ashokkumarkannan1** [comment]
Classifying this as layer-wise PCC accumulation (isolated layers/slices pass; full depth fails). Same bucket as [#2413](https://github.com/tenstorrent/tt-xla/issues/2413), [#4725](https://github.com/tenstorrent/tt-xla/issues/4725), [#3786](https://github.com/tenstorrent/tt-xla/issues/3786). Needs a shared tool/approach for accumulation-class drops, will revisit when that tooling is ready.
